### PR TITLE
PILOT-1880/1980 Add new file status update endpoints

### PR DESCRIPTION
**Pull request by erikdvlp**
## Summary

- Add two new `task-stream` endpoints for managing file status updates in Redis.
    - These will eventually replace the old `task-dispatch` endpoints.
    - `POST` endpoint creates a new file status entry in Redis Streams.
    - `GET` endpoint retrieves entries from Redis Streams and returns them via an SSE connection.
- Update the local Docker setup to be easier to use.

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1880
https://indocconsortium.atlassian.net/browse/PILOT-1980

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Testing

Are there any new or updated tests to validate the changes?

- [x] Yes

## Test Directions

1. Review the 2 new endpoints in `dataops/routes/api_task_stream/task_stream.py`.
    1. Are these sufficient for writing all upload, download, copy, and delete records to Redis?
    2. Can these completely replace the `task-dispatch` endpoints? Or is there some functionality missing?
    3. Is there any reason that we would need update/delete endpoints right now? My thoughts are: update is not needed because we'll append new records instead, and delete can come later when we figure out how to auto-trim old entries. But I know these exist in `task-dispatch`.
4. Review the old code in `dataops/routes/api_file_operations/copy_dispatcher.py` and `.../delete_dispatcher.py`.
    1. I didn't touch these. Should I?
    2. There are Redis records being written in there. Do these need to be updated to write to Streams?

### CODE REVIEW

**Comment by QXgu** [comment]
- Are these sufficient for writing all upload, download, copy, and delete records to Redis?
      To my understanding, we record these file actions to display the status in the file panel. We need to tell users how is the status of the file actions they triggered.

- Can these completely replace the task-dispatch endpoints? Or is there some functionality missing?

       The copy, delete pipelines call PUT task-dispatch endpoints to update the job status once the copy and delete pipeline finished in order to show whether the copy/delete is succeed or failed. Also upload and download services are calling the POST to create redis record. (More detail might need to confirm with Zhiren)

- Is there any reason that we would need update/delete endpoints right now? My thoughts are: update is not needed 
because we'll append new records instead, and delete can come later when we figure out how to auto-trim old entries. But I know these exist in task-dispatch.
     Pipeline service is calling the Update endpoint. Delete is not used in pipeline service

**Comment by erikdvlp** [comment]
As a result of the comments above, I will make these changes:
1. Add a job ID field to the Redis entries to uniquely identify many records with the same job.
2. Add another GET endpoint that doesn't use SSE.
3. Update the copy and delete dispatchers to write to Streams.

File: dataops/routes/api_file_operations/copy_dispatcher.py [hunk 1]
```diff
@@ -49,10 +52,27 @@ async def execute(
         except ValueError as e:
             return EAPIResponseCode.bad_request, str(e)
 
+        target_names = []
+        target_type = 'batch'
+        for target in data.payload.targets:
+            item = await get_resource_by_id(target.id)
+            target_names.append(item['name'])
+            if len(data.payload.targets) == 1:
```
Comment (colorzzr) [inline dataops/routes/api_file_operations/copy_dispatcher.py:60]: would this be outside the for loop?

File: dataops/routes/api_file_operations/delete_dispatcher.py [hunk 2]
```diff
@@ -46,8 +49,25 @@ async def execute(
         except ValueError as e:
             return EAPIResponseCode.bad_request, str(e)
 
+        target_names = []
+        target_type = 'batch'
+        for target in data.payload.targets:
+            item = await get_resource_by_id(target.id)
+            target_names.append(item['name'])
+            if len(data.payload.targets) == 1:
```
Comment (colorzzr) [inline dataops/routes/api_file_operations/delete_dispatcher.py:57]: same here would this be outside the for loop?

File: dataops/routes/api_file_operations/copy_dispatcher.py (hunk 1 above)
Comment (erikdvlp) [reply dataops/routes/api_file_operations/copy_dispatcher.py]: This functions the way that it should, but it is hard to read, so I've rewritten it.

**Review by colorzzr — APPROVED** [review]

**Pull request merged**